include/net/can.h

```diff
@@ -0,0 +1,192 @@
+#define CAN_NET_IF_ADDR_DEST_POS   14
+#define CAN_NET_IF_ADDR_DEST_MASK  (CAN_NET_IF_ADDR_MASK << CAN_NET_IF_ADDR_DEST_POS)
+#define CAN_NET_IF_ADDR_SRC_POS    0
+
+#define CAN_NET_FILTER_NOT_SET -1
+
+/** @endcond */
+
+/*
+ *             +-----------+           +-----------+
+ *             |           |           |           |
+ *             |   IPv6    |           |   IPv6    |
+ *             |  6LoCAN   |           |  6LoCAN   |
+ *             |           |           |           |
+ *             +----+-+----+           +----+-+----+
+ *                  | |                     | |                   +---+
+ *  +----+          | |                     | |                  /     \   +-+
+ *  |    |          | |                     | |          +--+   /       \_/   \
+ * +++   +---+   +--+----+   +-------+   +--+----+   +--/    \_/               \
+ * | |        \ /     |   \ /         \ /     |   \ /  /                       |
+ * | |         X      |    X    CAN    X      |    X  |         Internet       |
+ * | |        / \     |   / \         / \     |   / \  \                       /
+ * +++   +---+   +----+--+   +-------+   +----+--+   +--+                     /
+ *  |    |                                               +-------------------+
+ *  +----+
+ */
+struct net_can_api {
+	/**
+	 * The net_if_api must be placed in first position in this
+	 * struct so that we are compatible with network interface API.
+	 */
+	struct net_if_api iface_api;
+
+	/** Send a single CAN frame */
+	int (*send)(struct device *dev, const struct zcan_frame *frame,
+		    can_tx_callback_t cb, void *cb_arg, s32_t timeout);
+	/** Attach a filter with it's callback */
+	int (*attach_filter)(struct device *dev, can_rx_callback_t cb,
+			     void *cb_arg, const struct zcan_filter *filter);
+	/** Detach a filter */
+	void (*detach_filter)(struct device *dev, int filter_id);
+	/** Enable or disable the reception of frames for net CAN */
+	int (*enable)(struct device *dev, bool enable);
+};
+
+/** @cond INTERNAL_HIDDEN */
+
+#define CANBUS_L2_CTX_TYPE	struct net_can_context *
+
+/**
+ * Context for canbus net device.
+ */
+struct canbus_net_ctx {
+	/** Filter ID for link layer duplicate address detection. */
+	int dad_filter_id;
+	/** Work item for responding to link layer DAD requests. */
+	struct k_work dad_work;
+	/** The interface associated with this device */
+	struct net_if *iface;
+	/** The link layer address chosen for this interface */
+	u16_t ll_addr;
+};
+
+/**
+ * Canbus link layer addresses have a length of 14 bit for source and destination.
+ * Both together are 28 bit to fit a CAN extended identifier with 29 bit length.
+ */
+struct net_can_lladdr {
+	u16_t addr : 14;
+};
+
+/*
+ * STmin is split in two ranges:
+ *   0-127: 0ms-127ms
+ * 128-240: Reserved
+ * 241-249: 100us-900us (multiples of 100us)
+ * 250-   : Reserved
+ */
+struct can_net_fc_opts {
+	u8_t bs;     /** Block size. Number of CF PDUs before next CF is sent */
```

Comment:
I think doxygen requires that comments are like this
`/**< Block size. Number of CF PDUs before next CF is sent */`
if the comment refers to things in the same line.